lib/chat/chat_messenger.dart

```diff
@@ -157,49 +158,63 @@ class ChatMessengerState extends State<ChatMessenger> {
     return Container(
```

Comment:
This whole return statement is kinda overwhelming, so it's probably better to break it into multiple widget functions?